ext/ldap/ldap.c

```diff
@@ -1028,7 +1167,7 @@ PHP_FUNCTION(ldap_connect)
 			size_t urllen = hostlen + sizeof( "ldap://:65535" );
 
 			if (port <= 0 || port > 65535) {
-				efree(ld);
+				zval_ptr_dtor(return_value);
```

Comment:
Don't destroy the return value here, it will cause a double free. `RETURN_THROWS()` doesn't overwrite return_value, so the return_value will be destroyed as usual.

Reply:
Oh, I didn't realize before that destroying is only needed when the return value is overwritten. Thanks for the explanation!